usefulUtilities/webAppTemplate/webRouter.js

```diff
@@ -0,0 +1,172 @@
+
+const express = require('express');
+const router = express.Router();
+const Database = require('./dbClass.js');
+const dbConfig = require("./dbInfo.json").config;
+const mysql = require('mysql');
+
+const connectionConfig = {
+    host: dbConfig.mySQLHost,
+    user: dbConfig.mySQLUsername,
+    password: dbConfig.mySQLPassword,
+    database: dbConfig.databaseName,
+} 
+
+
+// Use this if you need to handle a preflight request, if not, you can remove
+router.all("*", (req,res, next) => {
+    if (req.method === "OPTIONS") {
+        res.header('Access-Control-Allow-Origin', req.headers.origin);
+        res.setHeader('Access-Control-Allow-Methods', 'POST, OPTIONS');
+        res.statusCode = 200;
+        return res.end();
+    }
+    next();
+})
+
+
+// Adding vars for the below examples
+const TABLE_NAME = "TABLE_NAME"
+const PATH = "/path/";
+
+router.get("/", (req, res) => {
+    res.send("TEST")
+})
+
+
+// GET: Request a record
+router.get(PATH, (req, res) => {
+    let query = `SELECT * FROM \`${TABLE_NAME}\``
+
+    Database.execute(connectionConfig, 
+        database => database.query(query)
+        .then((results) => {
+            let responseObject = {
+                status: "success",
+                data: results
+            };
+            console.log(`Got the ${TABLE_NAME}, sending over!`);
+            res.statusCode = 200;
+            res.setHeader('Content-Type', 'application/json');
+            res.end(JSON.stringify(responseObject));
+        })
+        .catch(e => {
+            console.log("e", e)
+            let responseObject = {
+                status: "error",
+                text: "Error getting response"
```

Comment:
this (and other) error messages aren't exactly descriptive/accurate - but because this is only a template, i don't think that's a big deal :)